### PR TITLE
Add  KB3x keyboard

### PR DESCRIPTION
**Pull request by DOIO2022**
## Description

Add  KB3x  keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/doio/kb3x/readme.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+* Keyboard Maintainer: DOIO2022
+* Hardware Supported: DOIO Knob keypad
+
+Make example for this keyboard (after setting up your build environment):
+
+    make doio/kb19:default
+
+Flashing example for this keyboard:
+
+    make doio/kb19:default:flash
```
Comment (waffle87) [inline keyboards/doio/kb3x/readme.md:14]:
```suggestion
    make doio/kb3x:default

Flashing example for this keyboard:

    make doio/kb3x:default:flash
```

File: keyboards/doio/kb3x/readme.md
Comment (dunk2k) [inline keyboards/doio/kb3x/readme.md:1]: Image of this keypad would be helpful 😉

Comment (DOIO2022) [reply keyboards/doio/kb3x/readme.md:1]: @dunk2k  Thank you for your suggestion and I've added it.

File: keyboards/doio/kb3x/info.json
Comment (fauxpark) [inline keyboards/doio/kb3x/info.json:1]: Please run `qmk format-json -i` on this file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/doio/kb3x/info.json
Comment (DOIO2022) [reply keyboards/doio/kb3x/info.json:1]: @fauxpark  Hi ,I've run it. Thank you!

**Comment by zvecr** [comment]
image

All clicking that "update branch" button is doing is using up CI minutes and making the queue longer for other PRs. I suggest you actually implement the requested changes <https://github.com/qmk/qmk_firmware/pull/23015#discussion_r1501098124>.

File: keyboards/doio/kb3x/readme.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# doio/kb3x
+
+![kb3x](https://github.com/DOIO2022/qmk_firmware/assets/116554792/cc2d265c-75e4-46fe-b8a1-19698ae61a3e)
```
[the remark this answers is not in the document]
Comment (DOIO2022) [reply keyboards/doio/kb3x/readme.md]: @waffle87  Hi,I‘ve changed it. Thank you!

**Comment by DOIO2022** [comment]
> image
> 
> All clicking that "update branch" button is doing is using up CI minutes and making the queue longer for other PRs. I suggest you actually implement the requested changes [#23015 (comment)](https://github.com/qmk/qmk_firmware/pull/23015#discussion_r1501098124).

@zvecr   Hi，I have been troubled by this problem for a long time，could you tell me the mean of  "implement the requested changes"? I don't know what should I do next. Thank  you  for your help.

File: keyboards/doio/kb3x/readme.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# doio/kb3x
+
+[![kb3x-01](https://imgur.com/gallery/vVF4TNW)](https://imgur.com/gallery/vVF4TNW)
```
Comment (zvecr) [inline keyboards/doio/kb3x/readme.md:3]:
```suggestion
![kb3x-01](https://i.imgur.com/6DvIu9Ml.png)
```

**Review by fauxpark — APPROVED** [review]

**Comment by DOIO2022** [comment]
@fauxpark  Hi, I accidentally closed this pull. Can I open it again? Thank you!

**Comment by zvecr** [comment]
No, as you have force pushed while the PR was closed. (An action that was not requested or required)

**Pull request closed without merging**